### PR TITLE
aop: 13.5 bring up +  mics + als + generalize afk/epic

### PR DESCRIPTION
**Pull request by eiln**
aop_audio: stream hpai device (high power audio input) microphone samples from aop to admac. output seems to be 32-bit LE floats in groups of 3 (3 mic array)
aop_als: monitor/report red, green, blue, clear, lux, gain etc information from ambient light sensor (ALS)

Tested j293ap 13.5

Also includes afk/epic/rbep generalizations which hopefully don't break DCP

### CODE REVIEW

**Comment by eiln** [comment]
nvm this broke dcp. fixing it

File: proxyclient/m1n1/fw/afk/rbep.py [hunk 1]
```diff
@@ -50,19 +52,24 @@ class AFKEP_Shutdown(AFKEPMessage):
 class AFKEP_Shutdown_Ack(AFKEPMessage):
     TYPE = 63, 48, Constant(0xc1)
 
-
-class AFKError(Exception):
-    pass
+# TODO no one else but AOP uses AFKRingBuf right?
```
Comment (jannau) [inline proxyclient/m1n1/fw/afk/rbep.py:55]: DCP uses AFKRingBuf on some endpoints via `EPICEndpoint`. Used in experiments/dcpext_iboot.py

Comment (eiln) [reply proxyclient/m1n1/fw/afk/rbep.py]: New commits don't break DCP experiments. DCP/AOP AFKRingbuf are too different to share (e.g. different EPIC header, different ring size, AOP also writes ptrs to DART and etc), everything else is fine though. I think we can move afk to soc/apple/ and just have fn pointers for the rbep r/w calls

**Comment by eiln** [comment]
@jannau  I generalized AFK/EPIC code to both DCP and AOP. I feel like I should send these in a separate series but it obviously depends on the above. I think marcan's on holiday. From my tests (dcp/dcp_iboot; I can't test dptx/dcpext) it doesn't break DCP but I'd like to get your ack

**Comment by pbasista** [comment]
Hi, could someone familiar with Asahi Linux and Apple audio help to move this PR or more generally the microphone support forward?

**Pull request merged**